### DISCUSSION TITLE
How to apply a mask to an image with transparency, similar to Photoshop Layer Mask?

### DISCUSSION THREAD

**Discussion by hbCyber** [ImageMagick/ImageMagick#2992] [Help] [answered]
I have an image for which I have created a mask. The original image has transparency which I want to preserve. I cannot find the right command(s) to get the desired effect of applying the mask (i.e. black pixels from the mask make original image's pixels transparent) while also preserving the original image's transparency.

Original image:
![arcadeCps1](https://user-images.githubusercontent.com/7698618/102248286-fc183f80-3f3b-11eb-8f73-25e86b1e533e.png)

Mask:
![arcadeCps1-mask](https://user-images.githubusercontent.com/7698618/102248306-033f4d80-3f3c-11eb-9515-6ea4fd0a3ea5.png)

What I want (I created that image using a Layer Mask in Photoshop):
![arcadeCps1-desired](https://user-images.githubusercontent.com/7698618/102248364-10f4d300-3f3c-11eb-8ae6-e98ff4414199.png)

What I actually get with my command(s):
```
convert arcadeCps1.png arcadeCps1-mask.png -compose copy-opacity -composite arcadeC
ps1-inner.png
```
![arcadeCps1-inner](https://user-images.githubusercontent.com/7698618/102248496-2ec23800-3f3c-11eb-9e4b-06c035833d97.png)
i.e. the output includes a black background which didn't exist before in the original image.

What am I missing?

**Comment by snibgo** [comment, marked as answer]
"-compose copy-opacity -composite" copies the opacity (the alpha channel), totally overwriting the original opacity. But you want to _modify_ the opacity by the mask. For example, we can multiply them together.
```
magick capcom.png ^
  ( +clone -alpha extract capcom_mask.png -compose Multiply -composite ) ^
  -compose CopyOpacity -composite ^
  out.png
```
Windows syntax. Adjust for bash.

**Reply by fmw42** [reply]
Bash:

```
convert capcom.png \
\( -clone 0 -alpha extract mask.png -compose multiply -composite \) \
-alpha off -compose copy_opacity -composite \
capcom_proc.png

```

![capcom_proc](https://user-images.githubusercontent.com/13458218/102253434-2c1ffe00-3ebc-11eb-89ef-aa92abbcaac1.png)
